### PR TITLE
Some more flow types

### PR DESCRIPTION
**Pull request by PinkaminaDianePie**
add flow types to separators, icon as button and overlayed box components

### CODE REVIEW

File: src/components/icon-as-button/IconAsButton.jsx [hunk 1]
```diff
@@ -1,22 +1,52 @@
+// @flow
 import React from 'react';
-import PropTypes from 'prop-types';
+import type {Node} from 'react';
 import classNames from 'classnames';
-import Icon, {TYPE, ICON_COLOR} from '../icons/Icon';
+import * as IconModule from '../icons/Icon';
```
Comment (PinkaminaDianePie) [inline src/components/icon-as-button/IconAsButton.jsx]: `import as *` is only way to import types in a way compatible with our vs code intellisence

Comment (aju) [reply src/components/icon-as-button/IconAsButton.jsx]: Why not just as Icon ?

Comment (aju) [reply src/components/icon-as-button/IconAsButton.jsx]: ah to spread it 👍

File: src/components/icon-as-button/IconAsButton.jsx [hunk 2]
```diff
@@ -1,22 +1,52 @@
...
+  [SIZE.SMALL]: 18,
+  [SIZE.XSMALL]: 14,
+  [SIZE.XXSMALL]: 10
+};
+
+type IconAsButtonPropsType = {
+  size?: IconSizeType,
+  color?: IconModule.IconColorType,
+  type?: ?IconModule.IconTypeType,
+  children?: ?Element<any>,
```
Comment (maciekbb) [inline src/components/icon-as-button/IconAsButton.jsx]: hmm why not `React.Node`?

Comment (PinkaminaDianePie) [reply src/components/icon-as-button/IconAsButton.jsx]: We should pass exactly one element, not an array, and `Element` is just like that, while `Node` can be an array or literal. We had such test before, you can check it :)

**Pull request merged**